samples/matter/common/src/persistent_storage/persistent_storage.h

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#pragma once
+
+#include "persistent_storage_common.h"
+
+namespace Nrf
+{
+
+class PersistentStorageImpl;
+
+/**
+ * @brief Interface class to manage persistent storage using generic PersistentStorageNode data unit
+ */
+class PersistentStorage {
+public:
+	/**
+	 * @brief Load all data from persistent storage
+	 *
+	 * @return true if data has been loaded
+	 * @return false otherwise
+	 */
+	PSErrorCode Init();
+	/**
+	 * @brief Store data into the persistent storage
+	 *
+	 * @param node address of the tree node containing information about the key
+	 * @param data data to store into a specific key
+	 * @param dataSize a size of input buffer
+	 * @return true if key has been written successfully
+	 * @return false an error occurred
+	 */
+	PSErrorCode Store(PersistentStorageNode *node, const void *data, size_t dataSize);
+
+	/**
+	 * @brief Load data from the persistent storage
+	 *
+	 * @param node address of the tree node containing information about the key
+	 * @param data data buffer to load from a specific key
+	 * @param dataMaxSize a size of data buffer to load
+	 * @param outSize an actual size of read data
+	 * @return true if key has been loaded successfully
+	 * @return false an error occurred
+	 */
+	PSErrorCode Load(PersistentStorageNode *node, void *data, size_t dataMaxSize, size_t &outSize);
+
+	/**
+	 * @brief Check if given key entry exists in the persistent storage
+	 *
+	 * @param node address of the tree node containing information about the key
+	 * @return true if key entry exists
+	 * @return false if key entry does not exist
+	 */
+	PSErrorCode HasEntry(PersistentStorageNode *node);
+
+	/**
+	 * @brief Remove given key entry from the persistent storage
+	 *
+	 * @param node address of the tree node containing information about the key
+	 * @return true if key entry has been removed successfully
+	 * @return false an error occurred
+	 */
+	PSErrorCode Remove(PersistentStorageNode *node);
+
+	/* Secure storage API counterparts.*/
+	PSErrorCode InitSecure();
+	PSErrorCode StoreSecure(PersistentStorageNode *node, const void *data, size_t dataSize);
+	PSErrorCode LoadSecure(PersistentStorageNode *node, void *data, size_t dataMaxSize, size_t &outSize);
+	PSErrorCode HasEntrySecure(PersistentStorageNode *node);
+	PSErrorCode RemoveSecure(PersistentStorageNode *node);
```

Comment:
Why do we need these `_secure` methods? I can see that you used the static polymorphism here, and the methods can be the same for secure and insecure backends. It is enough to use the proper instance, isn't it?

Reply:
discussed f2f, these secure and non-secure backends are not mutually exclusive.